Apps/DBMaintainer/Apps/BCPProvDrugDBApp.cs

```diff
@@ -183,13 +185,13 @@ private void ProcessDownload(string pharmaCareDrugSourceFolder, FileDownload pha
             this.RemoveOldFiles(pharmaCareDrugFileDownload);
 
             // Add new pharma care drug file download to database
-            this.AddFileToDb(pharmaCareDrugFileDownload);
+            await this.AddFileToDbAsync(pharmaCareDrugFileDownload, ct);
 
             // Remove old pharmacy assessment file download in database
             this.RemoveOldFiles(pharmacyAssessmentFileDownload);
 
             // Add new pharmacy assessment file download to database
-            this.AddFileToDb(pharmacyAssessmentFileDownload);
+            await this.AddFileToDbAsync(pharmacyAssessmentFileDownload, ct);
 
             this.Logger.LogInformation("Parsing Provincial PharmaCare Drug file");
             IList<PharmaCareDrug> pharmaCareDrugs = this.Parser.ParsePharmaCareDrugFile(pharmaCareDrugFile, pharmaCareDrugFileDownload);
```

Comment:
Suggest converting ParsePharmaCareDrugFile in IPharmaCareDrugParser to async using CsvReader's GetRecordsAsync.